kotlinpoet/src/main/java/com/squareup/kotlinpoet/PropertySpec.kt

```diff
@@ -54,6 +58,13 @@ public class PropertySpec private constructor(
     require(mutable || setter == null) {
       "only a mutable property can have a setter"
     }
+    if (contextReceiverTypes.isNotEmpty()) {
+      if (mutable) {
+        requireNotNull(getter) { "mutable properties with context receivers require a $GETTER" }
+        requireNotNull(setter) { "mutable properties with context receivers require a $SETTER" }
+      }
+      requireNotNull(getter) { "immutable properties with context receivers require a $GETTER" }
```

Comment:
This feels a bit cleaner and avoids duplication:
```suggestion
      requireNotNull(getter) { "properties with context receivers require a $GETTER" }
      if (mutable) {
        requireNotNull(setter) { "mutable properties with context receivers require a $SETTER" }
      }
```

Reply:
Note that this change will require tests to be updated as well.

Reply:
> Note that this change will require tests to be updated as well.

Which is obviously something I forgot. Updated them just now.